wagtail/contrib/forms/tests/test_views.py

```diff
@@ -158,10 +159,15 @@ class TestFormsIndex(WagtailTestUtils, TestCase):
 
     def setUp(self):
         self.login(username="siteeditor", password="password")
+        get_form_types.cache_clear()
 
     @classmethod
     def setUpTestData(cls):
         cls.form_page = Page.objects.specific().get(url_path="/home/contact-us/")
+        # Save a revision so latest_revision_created_at is populated, and thus
+        # the form page is always shown first when using the default ordering
+        # (latest_revision_created_at descending).
+        cls.form_page.save_revision()
```

Comment:
The tests were failing on MariaDB but not on MySQL – so it wasn't caused by how MySQL/MariaDB FTS only works with committed data.

It was because existing tests relied on the fact that this specific form page appears in the first page of the pagination. I refactored this test class so that the `make_form_pages()` method is called once in `setUpTestData()`, to speed it up instead of calling it in each test method where we need it.

The page listing's default ordering is `-latest_revision_created_at`, with `nulls_last=True`. None of the form pages in this test, whether they come from fixtures or created by `make_form_pages()`, have the `latest_revision_created_at` field set (they're all null). On databases other than MariaDB, it seems to consistently return the results in such a way that the "Contact us" appears in the first page. On MariaDB, it fails, because it probably gets shoved down to the last page (or just indeterministic).

Calling `save_revision()` ensures this "Contact us" page has `latest_revision_created_at` set to `now()`, thus always appears first.